case-study.Rmd

```diff
@@ -0,0 +1,467 @@
+active in this dataset.
+
+```{r}
+pathway_limma <- build_limma(pathway_activity, phenotypes = data_pheno,
+                             control_pheno = "wt", use.bonferroni = TRUE)
+# combine pathway association and pathway activation test results
+combined_result <- combine_geneset_outputs(
+  signature_geneset_association = pathway_association,
+  geneset_limma_result = pathway_limma)
+knitr::kable(combined_result, digits = 4, row.names = FALSE, align = "c")
+```
+
+### Gene-gene networks
+We next visualize how genes in the active signatures cluster in the
+ADAGE-derived gene-gene network.
+
+We calculate an expression fold change for each gene and pass it to the
+gene-gene network to show as node color. Again, we use limma to test
+differential expression and get the logFC.
+
+```{r}
+data_raw_limma <- build_limma(input_data = data_raw, phenotypes = data_pheno,
+                              control_pheno = "wt")
+# build a gene:fold change table from limma result
+gene_logFC <- data.frame(geneID = rownames(data_raw_limma),
+                         logFC = data_raw_limma$logFC)
+```
+
+Visualize the ADAGE gene-gene network of the active signatures.
+
+```{r}
+visualize_gene_network(selected_signatures = unique_active_sigs,
+                       gene_color_value = gene_logFC,
+                       model = model, cor_cutoff = 0.5,
+                       curated_pathways = KEGG)
+```
+
+We can also visualize one signature at a time. 
+
+```{r}
+visualize_gene_network(selected_signatures = "Node35pos",
+                       gene_color_value = gene_logFC,
+                       model = model, cor_cutoff = 0.5,
+                       curated_pathways = KEGG)
+```
+
+# A comparison between ADAGEpath and GSEA
+We also perform a GSEA analysis on this dataset.
+
+```{r}
+# specifiy file path to the GSEA java program
+GSEA_file <- "gsea2-2.2.4.jar"
+# prefix of the folder that stores GSEA results
+GSEA_output_name <- "GSEA_result"
+# folder to store GSEA data and results
+GSEA_folder <- "GSEA_data"
+
+# NOTICE!!! The folder cannot already exist.
+if (dir.exists(GSEA_folder)) {
+  stop("The GSEA folder already exists. Please give a new folder name.")
+}
+dir.create(GSEA_folder, showWarnings = TRUE)
+
+
+if (file.exists(GSEA_file)){
+
+  # create the class lable file for gsea
+  label_file <- file.path(GSEA_folder, "label.cls")
+  write(paste(length(data_pheno), length(unique(data_pheno)), 1, sep = " "),
+        file = label_file)
+  write("# con1 con2", file = label_file, append = TRUE)
+  write(paste(data_pheno, collapse = " "), label_file, append=TRUE)
+  
+  # create the gene expression file for gsea
+  expression_file <- file.path(GSEA_folder, "expression.txt")
+  readr::write_tsv(data_raw, expression_file, col_names = TRUE)
+
+  # create the gene set file for gsea
+  geneset_file <- file.path(GSEA_folder, "geneset.gmt")
+  write(paste(names(KEGG_subset)[1], "na", paste(unlist(KEGG_subset[[1]]),
```

Comment:
this is tough to understand. What does this file end up looking like?

Can you split out some of the functions?